src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -123,23 +123,23 @@ internal void AddPropertiesUsingSourceGenInfo()
             {
                 if (typeof(T) == typeof(object))
                 {
-                    return;
+                    return null;
                 }
 
                 if (PropertyInfoForTypeInfo.ConverterBase.ElementType != null)
                 {
                     // Nullable<> or F# optional converter's strategy is set to element's strategy
-                    return;
+                    return null;
```

Comment:
Are there some `Debug.Assert`s that would be relevant to verify that this is a `Nullable` or F# optional?

Reply:
This specific check is happening because `Nulalble<T>` converter currently reports `Object` as its converter strategy but it actually never uses properties - IMO this is incorrect but I didn't want to do the needed refactoring to change - I was preserving existing behavior here. I'll refrain from doing this change in this PR

Reply:
> This specific check is happening because Nulalble<T> converter currently reports Object as its converter strategy but it actually never uses properties

It only will report `Object` if `T` is a struct. In any case it would make sense to an assertion here in case the `ElementType` assumption gets invalidated by a different converter in the future.